terracumber/terraformer.py

```diff
@@ -56,7 +57,12 @@ def inject_repos(self, custom_repositories_json):
                     replacement = ''.join(replacement_list)
                     placeholder = '//' + node + '_additional_repos'
                     for line in fileinput.input("%s/main.tf" % self.terraform_path, inplace=True):
-                        print(line.rstrip().replace(placeholder, replacement))
+                        (new_line, n) = subn(placeholder, replacement, line)
```

Comment:
Are you sure `line.rstrip()` is not needed? That's removing the whitespaces at the end of the line: https://docs.python.org/3.6/library/stdtypes.html#bytes.rstrip

Reply:
It's mostly used to remove the linefeed.

Instead of removing the linefeed with `rstrip()` and re-adding it with `print()`, I simply tell `print()` to not add it in the first place with parameter `end = ''`. More efficient and straightforward.

Reply:
According to the doc: `If omitted or None, the chars argument defaults to removing ASCII whitespace.`

Reply:
So yeah, of course, the "newline" character is included, but some others as well.

However, if you are 100% that cleanup is will not needed, it's OK to have it this way.

Reply:
There's no reason to remove extra spaces if any. And I'm pretty sure the only reason for this `rstrip()` was only the linefeed.

I have checked in the rest of the code, you consistently use `end = ''` to avoid extra linefeeds.

Reply:
Do we? I only see one match.

But anyway, OK, if you don't need to cleanup more stuff, this should be ok.

Reply:
right, only one in addition to mine

still, that's consistent :tongue: .